### PR TITLE
wire up cancellation

### PR DESCRIPTION
**Pull request by buzzdecafe**
ping @avaq

is this more ore less what you had in mind? AFAICT `abort` is not wired to error events.

### CODE REVIEW

File: test/flapjax.test.js [hunk 1]
```diff
@@ -11,35 +11,23 @@
...
-    it('takes options and returns a Task', function() {
-      var task = flapjax(maria);
-      expect(task).to.be.an.object;
+    it('takes options and returns a Future', function() {
+      var future = flapjax(maria);
+      expect(future).to.be.an.object;
       ['fork', 'cleanup', 'ap', 'bimap', 'cata', 'chain', 'concat',
        'empty', 'fold', 'map', 'of', 'orElse', 'rejected',
        'rejectedMap', 'swap'].forEach(function(method) {
         expect(task[method]).to.be.a.function;
```
Comment (Avaq) [inline test/flapjax.test.js:25]: Is this passing? It seems the variable `task` is undefined, and even if it weren't many of these properties are not defined on Future instances.

File: test/flapjax.test.js [hunk 2]
```diff
@@ -11,35 +11,23 @@
...
-      flapjax(maria, dec).fork(
-        function(x) {
-          expect(x.withCredentials).to.be.true;
-          done();
-        },
-        function(x) {
-          expect(x.withCredentials).to.be.true;
-          done();
-        });
+    it('', function() {
```
Comment (Avaq) [inline test/flapjax.test.js:29]: An unnamed test testing for nothing?

File: src/flapjax.js [hunk 3]
```diff
@@ -53,6 +53,12 @@ export default function flapjax(opts) {
     xhr = setEvents(opts.events || {}, xhr);
     xhr.addEventListener('readystatechange', onStateChange(xhr), false);
     xhr.send();
+    
+    return () => {
+      if (xhr.readyState < xhr.DONE) {
+        xhr.abort();
+      }
+    };
```
Comment (Avaq) [inline src/flapjax.js:61]: Yep. This is what I meant. :)

**Review by Avaq — COMMENTED** [review]
LGTM, except for some concerns about the tests, but maybe I shouldn't meddle.

File: test/flapjax.test.js (hunk 2 above)
Comment (buzzdecafe) [reply test/flapjax.test.js:29]: _mea culpa_. I have been derelict on the testing, it's on the bottom of my priority list

File: test/flapjax.test.js (hunk 1 above)
Comment (buzzdecafe) [reply test/flapjax.test.js:25]: yes, the tests are a mess.

File: test/flapjax.test.js (hunk 2 above)
Comment (Avaq) [reply test/flapjax.test.js:29]: That's fine, fix the tests when you wish, or don't. I was only passing by and saw an opportunity to educate about cancellation in Fluture ;)

Comment (buzzdecafe) [reply test/flapjax.test.js:29]: Thanks! I appreciate you taking the time

**Pull request merged**